engine/schema/src/com/cloud/dc/dao/DataCenterDao.java

```diff
@@ -35,7 +35,7 @@
 
     String[] getNextAvailableMacAddressPair(long id, long mask);
 
-    Pair<String, Long> allocatePrivateIpAddress(long id, long podId, long instanceId, String reservationId);
+    Pair<Pair<String, Long>, Integer> allocatePrivateIpAddress(long id, long podId, long instanceId, String reservationId, boolean forSystemVms);
```

Comment:
can't we make a Pair<Pair<String, Long>, Integer> a
class allocationData extends Pair<Pair<String, Long>, Integer> {}
this is starting to read a bit like bad proza instead of poetry.